tensorflow/compiler/xla/service/pattern_matcher.h

```diff
@@ -2157,6 +2221,19 @@ XLA_NULLOP_PATTERN(PartitionId)
 XLA_NULLOP_PATTERN(ReplicaId)
 #undef XLA_NULLOP_PATTERN
 
+// Matches if the instruction is unary and has any of the given opcodes or is
+// the operand.
```

Comment:
This description is confusing because there are multiple operands in play here.  This does not communicate the meaning of the function to me -- I cannot parse what it means.

In addition, this suggests that we match either (a) one of the opcodes, or (b) the `arg` pattern.  But in fact that's not correct.  In case (a), then the operand of the unary op must match (b) the `arg` pattern.

This is why I find calling it OptionalUnaryOp to be more clear.  Then we could also write comments that think of this as an "optional op" as well.

Similarly above with the comment and function name`WithOpcodeOrOperand`.

Reply:
The implicit requirement of the operand of a unary op matching the `arg` pattern is standard behavior, though. Therefore, the first interpretation of case (a) isn't actually expected.